__tests__/app.test.js

```diff
@@ -2,14 +2,98 @@ const pool = require('../lib/utils/pool');
 const setup = require('../data/setup');
 const request = require('supertest');
 const app = require('../lib/app');
+const UserService = require('../lib/services/UserService');
 
-describe('backend-express-template routes', () => {
+const testUser = {
+  first: 'James',
+  last: 'Bond',
+  email: '1@1.com',
+  password: '123456',
+};
+
+const registerAndLogin = async (userTestLogin = {}) => {
+  const password = userTestLogin.password ?? testUser.password;
+  const agent = request.agent(app);
+  const user = await UserService.create({ ...testUser, ...userTestLogin });
+  const { email } = user;
+  await agent.post('/api/v1/users/sessions').send({ email, password });
+  return [agent, user];
+};
+
+describe('top-secret test bed', () => {
   beforeEach(() => {
     return setup(pool);
   });
-  it('example test - delete me!', () => {
-    expect(1).toEqual(1);
+  it('Test to Create a new user', async () => {
+    const resp = await request(app).post('/api/v1/users').send(testUser);
+    const { first, last, email } = testUser;
+
+    expect(resp.body).toEqual({
+      id: expect.any(String),
+      first,
+      last,
+      email,
+    });
+  });
+
+  it('check to confirm that current user exists', async () => {
+    const [agent, user] = await registerAndLogin();
+    const me = await agent.get('/api/v1/users/me');
+
+    expect(me.body).toEqual({
+      ...user,
+      exp: expect.any(Number),
+      iat: expect.any(Number),
+    });
+  });
+
+  it('Test to confirm a 401 error when trying to list all users', async () => {
+    const res = await request(app).get('/api/v1/users');
+
+    expect(res.body).toEqual({
+      message: 'Please sign in to continue',
+      status: 401,
+    });
+  });
+
+  it('Test to confirm a 403 error when signed in but do not have Admin privileges', async () => {
+    const [agent] = await registerAndLogin();
+    const res = await agent.get('/api/v1/users');
+    expect(res.body).toEqual({
+      message: 'You are not authorized to view this page',
+      status: 403,
+    });
+  });
+
+  it('Test that returns a list of users when signed in as admin', async () => {
+    const [agent, user] = await registerAndLogin({ email: 'admin' });
+    const res = await agent.get('/api/v1/users');
+    expect(res.body).toEqual([{ ...user }]);
+  });
+
+  it('test to list secret data', async () => {
+    const resp = await request(app).get('/api/v1/secrets');
+
+    expect(resp.body).toEqual({
+      message: 'Please sign in to continue',
+      status: 401,
+    });
```

Comment:
Be sure to add in a test that verifies we receive secrets after signing in, as this test is more so testing that the authentication middleware is working properly.